llvm/utils/TableGen/SearchableTableEmitter.cpp

```diff
@@ -115,20 +115,20 @@ class SearchableTableEmitter {
 
   std::string primaryRepresentation(SMLoc Loc, const GenericField &Field,
                                     const Init *I) {
-    if (const StringInit *SI = dyn_cast<StringInit>(I)) {
+    if (const auto *SI = dyn_cast<StringInit>(I)) {
```

Comment:
While we're here should we remove most of the `else`'s since the `if` body already returned?